torch/utils/data/_utils/collate.py

```diff
@@ -158,6 +158,11 @@ def collate_tensor_fn(batch, *, collate_fn_map: Optional[Dict[Union[Type, Tuple[
             "Batches of nested tensors are not currently supported by the default collate_fn; "
             "please provide a custom collate_fn to handle them appropriately."
         )
+    if elem.layout in {torch.sparse_coo, torch.sparse_csr, torch.sparse_bsr, torch.sparse_csc, torch.sparse_bsc}:
+        raise RuntimeError(
+            "Batches of sparse tensors are not currently supported by the default collate_fn; "
+            "please provide a custom collate_fn to handle them appropriately."
```

Comment:
Do users commonly know how to handle sparse tensors in a custom collate_fn?

Reply:
> Do users commonly know how to handle sparse tensors in a custom collate_fn?

I think there is no definite answer. However, I trust that when users work with sparse tensors, they also are skilled to implement a custom collate_fn that meets their application requirements re how to stack batches of sparse tensors together. Some options are listed below.

IIUC, in terms of strided tensors, the purpose of collate function is to stack the strided batches together while users are provided a way to define their own stacking method.

In terms of sparse tensors, there also exists multiple approaches to stack batches of sparse tensors. For instance, stacking of COO tensors could mean stacking of indices and values to form a tensor with higher sparse dimension than its batches have. Or, if the indices of all COO batches are the same, one would want to stack only the values to form a hybrid tensor with extra dense dimensions.

Similarly, there exists multiple stacking methods for batches of sparse compressed tensors. For instance, one could create a sparse compressed tensor with a batch dimension, or if indices of all batches match, one could stack only the values by introducing an extra dense dimension.